driver/connect.c

```diff
@@ -17,6 +17,13 @@
 #include "util.h"
 #include "dsn.h"
 
+/* to access Curl_base64_decode(), not part of cURL's API */
+#ifndef CURL_STATICLIB
+#error "dynamically linked cURL library is not supported"
+#else
+#include "../lib/curl_base64.h"
```

Comment:
It seems a bit dodgy to be using a private implementation detail of libcurl rather than the public API.  I guess it's nice in that it avoids linking another DLL, but it could be broken by future changes to libcurl.  An alternative that's available with out-of-the-box Windows is `CryptStringToBinary`.  But obviously using that would be less portable to Linux if we ever wanted to do that.

Reply:
Thanks.
It is indeed not great to use the private function.
I just thought that since (1) the private Base64 API in cURL hasn't significantly changed since 2011, (2) a change should be easily detected, (3) upgrades of libcurl are not too often and "manually" done and (4) libcurl will remain part of the code  across potentially multiple platforms, this could be a "cheap" and portable solution, at least for now.